pyaedt/application/Variables.py

```diff
@@ -267,18 +269,17 @@ def decompose_variable_value(variable_value, full_variables={}):
             float_value = float(variable_value)
         except ValueError:
             # search for a valid units string at the end of the variable_value
-            loc = re.search("[a-z_A-Z]+$", variable_value)
+            loc = re.search("[a-z_A-Z]+", variable_value)
             units = _find_units_in_dependent_variables(variable_value, full_variables)
 
             if loc:
                 loc_units = loc.span()[0]
                 extract_units = variable_value[loc_units:]
-                if unit_system(extract_units):
```

Comment:
@maxcapodi78  This `unit_system()` method is using the dictionary `AEDT_UNITS`. This dictionary does not contain units such as area m², volume m^3. Consequently it was not working for "3.123456m2" as described in issue #609.

Reply:
got it. Do you think we should add it or your fix handles all conditions?